gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -185,6 +185,34 @@ fixObject(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectp
 	}
 }
 
+struct HeapSizes {
+	uintptr_t freeBytes;
+	uintptr_t objectBytes;
+};
+
+static void
+clearFreeEntry(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectptr_t object, void *userData)
+{
+	MM_GCExtensionsBase *extensions = MM_GCExtensionsBase::getExtensions(omrVMThread->_vm);
+	MM_ParallelGlobalGC *collector = (MM_ParallelGlobalGC *)extensions->getGlobalCollector();
+
+	if (extensions->objectModel.isDeadObject(object)) {
+		if (extensions->objectModel.isSingleSlotDeadObject(object)) {
+			((HeapSizes *)userData)->freeBytes += extensions->objectModel.getSizeInBytesSingleSlotDeadObject(object);
+		} else {
+			/* We only clear both linked and unlinked free entries.
+			 */
+			OMRZeroMemory((void *)((uintptr_t)object + sizeof(MM_HeapLinkedFreeHeader)), extensions->objectModel.getSizeInBytesMultiSlotDeadObject(object) - sizeof(MM_HeapLinkedFreeHeader));
+			((HeapSizes *)userData)->freeBytes += extensions->objectModel.getSizeInBytesMultiSlotDeadObject(object);
+		}
+	} else {
+		/* We don't care for marked objects other than being able to assert
+		 * that we cleared anything that is not marked
```

Comment:
actually expand this comment:

 We don't care for marked objects other than being able to assert:
 - that we cleared anything that is not a marked object
 - we visited all items of the heap (that sum of sizes of all marked objects and holes is equal to heap size)